src/net.cpp

```diff
@@ -1270,9 +1348,17 @@ bool V2Transport::SetMessageToSend(CSerializedNetMsg& msg) noexcept
 
     if (m_send_state != SendState::APP_READY) return false;
     // Construct contents (encoding message type + payload).
-    std::vector<uint8_t> contents(13 + msg.data.size(), 0);
-    std::copy(msg.m_type.begin(), msg.m_type.end(), reinterpret_cast<char*>(contents.data() + 1));
-    std::copy(msg.data.begin(), msg.data.end(), contents.begin() + 13);
+    std::vector<uint8_t> contents;
+    auto short_message_id = V2_MESSAGE_MAP(msg.m_type);
+    if (short_message_id) {
+        contents.resize(1 + msg.data.size());
+        contents[0] = *short_message_id;
+        std::copy(msg.data.begin(), msg.data.end(), contents.begin() + 1);
+    } else {
+        contents.resize(13 + msg.data.size());
```

Comment:
nit: though logically identical with the current code, could be explicit about the fill value of the resize (in contrast to the short message case above, not all of the new elments after the resize are overwritten):
```suggestion
        contents.resize(13 + msg.data.size(), 0);
```